src/compiler/scala/tools/cmd/CommandLineParser.scala

```diff
@@ -35,7 +35,15 @@ object CommandLineParser {
   }
   private object DoubleQuoted extends QuotedExtractor('"')
   private object SingleQuoted extends QuotedExtractor('\'')
-  private val Word = """(\S+)(.*)""".r
+  object Word {
+    private val regex = """(\S+)""".r
+    def unapply(s: String): Option[(String, String)] = {
+      regex.findPrefixOf(s) match {
+        case Some(prefix) => Some(prefix, s.substring(prefix.length, s.length()))
```

Comment:
I've been puzzling over this comment. Could you help a human out? Am I being too nitpicky?

Not sure which smiley I'll get for this one, but I just noticed there's an overload of substring that only takes a `beginIndex`.